File: app/src/main/java/com/ignis/android_cleanarchitecture/data/util/VersionUtils.java
package com.ignis.android_cleanarchitecture.data.util;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;

/**
 * バージョン関連のUtilクラス
 * <p/>
 * Created by tamura_k on 2016/03/18.
 */
public class VersionUtils {

    /**
     * 自身のバージョン名を取得する
     *
     * @param context
     * @return
     */
    public static String getVersionName(Context context) {
        try {
            PackageInfo packageInfo = context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
            return packageInfo.versionName;
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
            return null;
        }
    }

    /**
     * 自身のバージョンコードを取得する
     *
     * @param context
     * @return
     */
    public static int getVersionCode(Context context) {
        try {
            PackageInfo packageInfo = context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
            return packageInfo.versionCode;
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
            return -1;
        }
    }

}

File: data/src/main/java/com/kazakago/cleanarchitecture/data/util/VersionUtils.kt
package com.kazakago.cleanarchitecture.data.util

import android.content.Context
import androidx.core.content.pm.PackageInfoCompat
import com.kazakago.cleanarchitecture.domain.model.about.VersionCode
import com.kazakago.cleanarchitecture.domain.model.about.VersionName

object VersionUtils {

    fun getVersionName(context: Context): VersionName {
        val packageInfo = context.packageManager.getPackageInfo(context.packageName, 0)
        return VersionName(packageInfo.versionName)
    }

    fun getVersionCode(context: Context): VersionCode {
        val packageInfo = context.packageManager.getPackageInfo(context.packageName, 0)
        return VersionCode(PackageInfoCompat.getLongVersionCode(packageInfo))
    }

}
